views/urls.go

```diff
@@ -3,15 +3,63 @@ package views
 import (
 	"net/url"
 
+	"github.com/a-h/templ"
 	"github.com/ugent-library/bind"
 )
 
-func urlWithQuery(u *url.URL, v any) *url.URL {
-	vals, err := bind.EncodeQuery(v)
+type URLBuilder struct {
+	url *url.URL
+}
+
+func NewURLBuilder(base string) *URLBuilder {
+	url, err := url.Parse(base)
+	if err != nil {
+		// TODO: how to deal with this
+		panic(err)
+	}
```

Comment:
Not sure how to deal with this. This intended to be used in views, so can't really return an error as second return value. But apparently `url.Parse` is very forgiving on what you provide as the base url so probably won't happen any time soon.

Reply:
2 ways to think about this:
- we consider the wrong input a programming error, then panic is fine
- we consider the url as user input, then we need to handle the error

I think panic fine here

if you would like to handle the error:
templ, just like html/template allows a function to return `(string, error)`
so you could keep the error in an error struct field, and return at the end of the builder chain in `URL()`, `String()` and `SafeURL()`

Reply:
@nics I just considered another option. How about accepting a `*url.URL` as base parameter here instead of `string`? Then handling the `url.Parse` error is out of scope of the builder and in most cases we'd be starting from a `*url.URL` anyway. Only in case of doing something like `NewURLBuilder(c.FrontendURL)...` you start from a string.

Reply:
good idea, or just have both

Reply:
Done